File: Sorting and Searching/9_Bubble_Sort.cpp

//Function to sort the array using bubble sort algorithm.
void bubbleSort(int arr[], int n)
{
    // Your code here    
        int i, j;
        //traverse the array upto the last second element
  for (i = 0; i < n - 1; i++)
    { 
      for (j = 0; j < n - i - 1; j++)
	{//when element at previous index is greater than element at next index
	  if (arr[j] > arr[j + 1])
	    {//swap those two elements
	      int temp = arr[j];
	      arr[j] = arr[j + 1];
	      arr[j + 1] = temp;
	    }
	}
    }

}

File: 9_Bubble-Sort.java
class Main{
    public static void main(String[] args) {
        int arr[] = {6,3,8,9,4,2};
        int n = arr.length;
        bubble_sort(arr,n);
        System.out.print("{");
        for(int i : arr){
            System.out.print(" "+i);
        }
        System.out.print(" }");
    }
    public static void bubble_sort(int arr[], int n){
        int temp;
        for(int i=0;i<n-1;i++){
            
            for(int j=0;j<n-i-1;j++){
                
                if(arr[j]>arr[j+1]){
                    temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                }
                
            }
            
        }
        
    }
}